### PR TITLE
Matt/java backend

### PR DESCRIPTION
**Pull request by mvb0005**
I don't think the protobuf is working.

### CODE REVIEW

File: src/main/java/com/auacm/database/model/Problem.java [hunk 1]
```diff
@@ -20,18 +20,21 @@
...
     private String difficulty;
 
+    @Transient
+    private String url;
+
     private long added;
 
     @Column(name = "comp_release")
-    private int competitionId;
+    private Integer competitionId;
```
Comment (Mac-Genius) [inline src/main/java/com/auacm/database/model/Problem.java:29]: If you can, leave this as an int.

File: src/main/java/com/auacm/database/model/Problem.java [hunk 2]
```diff
@@ -81,11 +84,11 @@ public void setAdded(long added) {
         this.added = added;
     }
 
-    public int getCompetitionId() {
+    public Integer getCompetitionId() {
```
Comment (Mac-Genius) [inline src/main/java/com/auacm/database/model/Problem.java:87]: Use int if you can.

File: src/main/java/com/auacm/database/model/Problem.java [hunk 3]
```diff
@@ -81,11 +84,11 @@ public void setAdded(long added) {
...
         return competitionId;
     }
 
-    public void setCompetitionId(int competitionId) {
+    public void setCompetitionId(Integer competitionId) {
```
Comment (Mac-Genius) [inline src/main/java/com/auacm/database/model/Problem.java:91]: Int please.

File: src/main/java/com/auacm/database/model/ProblemData.java [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
 @Table(name = "problem_data")
 public class ProblemData implements Serializable {
+
+    @JsonIgnore
     @Id
     private long pid;
 
     @Column(name = "time_limit")
-    private int timeLimit;
+    private Integer timeLimit;
```
Comment (Mac-Genius) [inline src/main/java/com/auacm/database/model/ProblemData.java:17]: Int here as well.

**Review by Mac-Genius — CHANGES_REQUESTED** [review]
Just change Integer to int. Everything else looks good and the problems populate on the problems page.

**Comment by mvb0005** [comment]
`competitionId` cannot be an int as long as there are null values in the database. I'm not sure how to change the default database from null to something more fitting like 0.

**Comment by Mac-Genius** [comment]
Alright, we can leave it as Integer for now then.

**Review by Mac-Genius — APPROVED** [review]

**Pull request merged**